src/smbh/log/core.clj

```diff
@@ -37,64 +40,58 @@
 
 (defmacro log-c
   ([method ctx]
-   (if (resolve '⠇⠕⠶⠻)
```

Comment:
here the resolve is called during expension time, not at runtime.

Reply:
The thing is that the expansion happens in the `smbh.log.core` namespace, and runtime, well, wherever the macro gets called.
This means that `(resolve '⠇⠕⠶⠻)` will try to get the braille-named variables' value at runtime, in the namespace where it gets called. But if one tries to evaluate this:
```clojure
(do ~(init-logger)
    ~(case (count args)
...)
```
it will call `(init-logger)` at the expansion time, and it will get executed in the `smbh.log.core` namespace, making it useless. If, instead, the tilde gets removed and the same call happens at the runtime - it will init the logger in the caller's namespace, as intended.

In fact, if I create an empty ns, paste there this:

```
(ns smbh.log.test
  (:require [smbh.log.core :as log]))

(log/warn-m "arst")
```
and try to load it with `...~(init-logger)`, it will throw this:
```
CompilerException java.lang.RuntimeException: Unable to resolve symbol: ⠇⠕⠶⠻ in this context, compiling:(/home/vandr0iy/proj/smbh-log/src/smbh/log/test.clj:4:1) 
```

Reply:
sorry for the verbose explanation 😅

Reply:
But, yeah - `(init-logger)` does a `defonce` only once per namespace, ending up as an a simple variable existence check every time a `(log-*)` gets called, afterwards. This shouldn't have any significant performance impact :)

Reply:
i am still confused about the resolve. in my old code, it works by resolving in the correct (current) namespace. as you can see, if i include the logger name in the output, i get this kind of logs
```
=> nil2017-12-07T21:47:54.986 INFO  nREPL-worker-2     - timeflow.worktime -  -  - hello

Loading src/dev/clj/stan_dev.clj... done
(log/info-m "hello")
2017-12-07T21:48:02.158 INFO  nREPL-worker-2     - stan-dev -  -  - hello
```

Reply:
it should be possible to resolve and create the var in the correct namespace at expansion time. at least `deflogger` can do that.